langdetect/utils/unicode_block.py

```diff
@@ -1,8 +1,459 @@
+    (UNICODE_IMPERIAL_ARAMAIC, 0x10840, 0x1085F),
+    (UNICODE_PHOENICIAN, 0x10900, 0x1091F),
+    (UNICODE_LYDIAN, 0x10920, 0x1093F),
+    (UNICODE_MEROITIC_HIEROGLYPHS, 0x10980, 0x1099F),
+    (UNICODE_MEROITIC_CURSIVE, 0x109A0, 0x109FF),
+    (UNICODE_KHAROSHTHI, 0x10A00, 0x10A5F),
+    (UNICODE_OLD_SOUTH_ARABIAN, 0x10A60, 0x10A7F),
+    (UNICODE_AVESTAN, 0x10B00, 0x10B3F),
+    (UNICODE_INSCRIPTIONAL_PARTHIAN, 0x10B40, 0x10B5F),
+    (UNICODE_INSCRIPTIONAL_PAHLAVI, 0x10B60, 0x10B7F),
+    (UNICODE_OLD_TURKIC, 0x10C00, 0x10C4F),
+    (UNICODE_RUMI_NUMERAL_SYMBOLS, 0x10E60, 0x10E7F),
+    (UNICODE_BRAHMI, 0x11000, 0x1107F),
+    (UNICODE_KAITHI, 0x11080, 0x110CF),
+    (UNICODE_SORA_SOMPENG, 0x110D0, 0x110FF),
+    (UNICODE_CHAKMA, 0x11100, 0x1114F),
+    (UNICODE_SHARADA, 0x11180, 0x111DF),
+    (UNICODE_TAKRI, 0x11680, 0x116CF),
+    (UNICODE_CUNEIFORM, 0x12000, 0x123FF),
+    (UNICODE_CUNEIFORM_NUMBERS_AND_PUNCTUATION, 0x12400, 0x1247F),
+    (UNICODE_EGYPTIAN_HIEROGLYPHS, 0x13000, 0x1342F),
+    (UNICODE_BAMUM_SUPPLEMENT, 0x16800, 0x16A3F),
+    (UNICODE_MIAO, 0x16F00, 0x16F9F),
+    (UNICODE_KANA_SUPPLEMENT, 0x1B000, 0x1B0FF),
+    (UNICODE_BYZANTINE_MUSICAL_SYMBOLS, 0x1D000, 0x1D0FF),
+    (UNICODE_MUSICAL_SYMBOLS, 0x1D100, 0x1D1FF),
+    (UNICODE_ANCIENT_GREEK_MUSICAL_NOTATION, 0x1D200, 0x1D24F),
+    (UNICODE_TAI_XUAN_JING_SYMBOLS, 0x1D300, 0x1D35F),
+    (UNICODE_COUNTING_ROD_NUMERALS, 0x1D360, 0x1D37F),
+    (UNICODE_MATHEMATICAL_ALPHANUMERIC_SYMBOLS, 0x1D400, 0x1D7FF),
+    (UNICODE_ARABIC_MATHEMATICAL_ALPHABETIC_SYMBOLS, 0x1EE00, 0x1EEFF),
+    (UNICODE_MAHJONG_TILES, 0x1F000, 0x1F02F),
+    (UNICODE_DOMINO_TILES, 0x1F030, 0x1F09F),
+    (UNICODE_PLAYING_CARDS, 0x1F0A0, 0x1F0FF),
+    (UNICODE_ENCLOSED_ALPHANUMERIC_SUPPLEMENT, 0x1F100, 0x1F1FF),
+    (UNICODE_ENCLOSED_IDEOGRAPHIC_SUPPLEMENT, 0x1F200, 0x1F2FF),
+    (UNICODE_MISCELLANEOUS_SYMBOLS_AND_PICTOGRAPHS, 0x1F300, 0x1F5FF),
+    (UNICODE_EMOTICONS, 0x1F600, 0x1F64F),
+    (UNICODE_TRANSPORT_AND_MAP_SYMBOLS, 0x1F680, 0x1F6FF),
+    (UNICODE_ALCHEMICAL_SYMBOLS, 0x1F700, 0x1F77F),
+    (UNICODE_CJK_UNIFIED_IDEOGRAPHS_EXTENSION_B, 0x20000, 0x2A6DF),
+    (UNICODE_CJK_UNIFIED_IDEOGRAPHS_EXTENSION_C, 0x2A700, 0x2B73F),
+    (UNICODE_CJK_UNIFIED_IDEOGRAPHS_EXTENSION_D, 0x2B740, 0x2B81F),
+    (UNICODE_CJK_COMPATIBILITY_IDEOGRAPHS_SUPPLEMENT, 0x2F800, 0x2FA1F),
+    (UNICODE_TAGS, 0xE0000, 0xE007F),
+    (UNICODE_VARIATION_SELECTORS_SUPPLEMENT, 0xE0100, 0xE01EF),
+    (UNICODE_SUPPLEMENTARY_PRIVATE_USE_AREA_A, 0xF0000, 0xFFFFF),
+    (UNICODE_SUPPLEMENTARY_PRIVATE_USE_AREA_B, 0x100000, 0x10FFFF),
+]
+
+NUM_BLOCKS = len(_unicode_blocks)
+
+
 def unicode_block(ch):
     '''Return the Unicode block name for ch, or None if ch has no block.'''
     cp = ord(ch)
+    # special case basic latin
+    if cp <= 0x7F:
+        return UNICODE_BASIC_LATIN
     # binary search for the correct block
-    be, en = 0, len(_unicode_blocks)-1
+    be, en = 0, NUM_BLOCKS
```

Comment:
Put there ``NUM_BLOCKS-1``. Otherwise we might get ``IndexError`` on line 459, when ``be`` converges ``en``.